include/aws/auth/credentials.h

```diff
@@ -211,17 +211,25 @@ struct aws_credentials_provider_x509_options {
     struct aws_client_bootstrap *bootstrap;
 
     /* TLS connection options that have been initialized with your x509 certificate and private key */
-    struct aws_tls_connection_options *tls_connection_options;
+    const struct aws_tls_connection_options *tls_connection_options;
```

Comment:
if you ever want to use this struct in another language like C++, const here will prohibit that.